bugfix: ensure live preview is not cached

nbosb opened the pull request:
I use a setup with a live preview showing an MJPEG-based stream created by Go2rtc during the countdown. When using an iPad to control the photobooth, the preview works fine when taking a picture. However, when I click the `newPhoto` button on the results page, only the last frame of the previous live stream is shown. Everything works fine when I use Firefox on Windows. I think this is exactly the same issue as described in #942.
It seems that Safari (iOS 17.7.10) and Chrome (141.0.7390.41) on the iPad cache the last frame when the `preview--ipcam` element is hidden, and only use this frame when it is shown again instead of fetching the stream.

<!--
    Thank you for contributing!
-->

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/PhotoboothProject/photobooth/blob/dev/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "x" next to an item)
<!-- Example:
- [x] Bug fix
-->

- [ ] Documentation update
- [x] Bug fix
- [ ] New feature
- [ ] Other, please explain:

<!--
    Please ensure your pull request is ready:

    - Please run 'npm run build' before submitting to have consistent formatting for both JavaScript & SCSS
    - Please run 'npm run eslint' to make sure there's no lint issues
-->

<!--
    The following is required for all pull requests:
-->

#### What changes did you make? (Give an overview)

It seems that caching cannot be disabled for background images applied to `div` elements.  
As a workaround, I added a cache-busting query parameter to the stream URL so that it’s unique each time the `preview--ipcam` element is shown.

To simplify handling the stream source, I changed the `preview--ipcam` element from a `div` with a CSS background image to an `<img>` element. This allows the stream URL to be passed directly via the `src` attribute instead of using `url('...')`.  
A timestamp (`t=<timestamp>`) is appended to the URL to ensure it’s different each time and therefore not cached by the browser.

##### Tested on
- Firefox 143.0.4 (Windows)
- Safari (iOS 17.7.10)
- Chrome 141.0.7390.41 (iOS 17.7.10)
- Chrome 141.0.7390.122 (Android 15)

##### Note
The stream URL must now be passed directly to the `<img>` element; the `url('...')` syntax is no longer needed.


#### Is there anything you'd like reviewers to focus on?
As I’m not a JS/PHP developer, I’d appreciate it if you could check whether I’ve missed any best practices or made any mistakes in the implementation. Please feel free to point out any issues with the code, especially if something seems unconventional or my implementation causes problems I am not aware of.

nbosb committed:
Ensure live preview is not cached

On some devices, the video preview freezes when the video is hidden and
then shown again (e.g, show preview during countdown; hide preview to
show taken picture; show preview during countdown of the next picture).
This is because the last frame can be cached by the browser when the
preview is hidden, and this cached frame is displayed when the preview
is shown again.
To disable caching, a unique parameter is added to the URL of the
stream of the stream.

Signed-off-by: Nils Bosbach <n.bosbach@gmx.de>

andi34 commented:
Hey and thanks for your pull request!
I've not tested yet, but looks good to me. The following files need to be adjusted too:
- admin/captureconfig.php
- admin/wgetcaptureconfig.php

Could you please adjust the config entries there too?

nbosb commented:
Hi, thanks for the hint. I have updated the files, too.

andi34 commented:
Thanks for the update. I am now thinking about how to handle already configured Photobooth to not break. Any ideas?

nbosb committed:
Migrate preview.url field in config

When the config has been created by a previous version, the preview.url
field may contain values wrapped in `url("...")`. The updated version no
longer requires this wrapper. If the preview.url field starts with `url(`
and ends with `)`, the `url(...)` wrapper and any surrounding quotes are
removed.

Signed-off-by: Nils Bosbach <n.bosbach@gmx.de>

nbosb commented on src/Service/ConfigurationService.php:
```diff
@@ -161,6 +161,11 @@ protected function processMigration(array $config): array
             $config['preview']['mode'] = 'device_cam';
         }
 
+        // Migrate Preview URL
+        if (isset($config['preview']['url']) && substr($config['preview']['url'], 0, 4) === 'url(' && substr($config['preview']['url'], -1) === ')') {
+            $config['preview']['url'] = trim(substr($config['preview']['url'], 4, -1), '"\'');
+        }
+
```
To handle migration of old configs, I added a check that removes the wrapping `url(...)` part from preview.url, including any leading or trailing quotes. If you prefer a different way, just let me know.

andi34 replied:
Thanks a lot :)
